src/models/Factor.js

```diff
@@ -244,20 +244,21 @@ function (Okta, Q, factorUtil, BaseLoginModel) {
             'delay': PUSH_INTERVAL
           };
           if (trans.status === 'MFA_CHALLENGE' && trans.poll) {
+            // In verify case since we poll we dont return the trans
+            // rather set it and continue to poll
+            setTransaction(trans);
             return Q.delay(PUSH_INTERVAL).then(function() {
               if (self.pushFactorHasAutoPush()) {
                 options.autoPush = function() {
                   return self.get('autoPush');
                 };
               }
-              return trans.poll(options);
+              return trans.poll(options).then(function(trans) {
+                setTransaction(trans);
+              });
             });
           }
-          return trans;
-        })
-        .fail(function (err) {
-          // Clean up the cookie on failure.
```

Comment:
I am curious - what is this for? Is it okay just to remove it?

Reply:
We have a fail case in BaseLoginModel https://github.com/okta/okta-signin-widget/pull/404/files#diff-733a4f9ae0ae362416e8e89d8ef73db7R54 which has other checks anyway  this feels it is unnecessary. I would add additional check https://github.com/okta/okta-signin-widget/pull/404/files#diff-733a4f9ae0ae362416e8e89d8ef73db7R42

Reply:
Also verified that this was a stale comment https://github.com/okta/okta-signin-widget/commit/7ad708db07845d89a8ad2cb84fac1d424b3a57e0#diff-14b6f8f7494d563336c6490bab399799R72